backend/glp/service.py

```
@@ -0,0 +1,800 @@
+        nowait=connection.features.has_select_for_update_nowait,
+    )
+    try:
+        return queryset.get(pk=1)
+    except GLPProgram.DoesNotExist:
+        raise GateError('GLP program does not exist yet.') from None
+    except OperationalError as exc:
+        # Map ONLY a lock-unavailable conflict to RunInProgress; anything else
+        # is a real database error and must propagate.
+        pgcode = getattr(getattr(exc, '__cause__', None), 'pgcode', None)
+        if pgcode == '55P03' or 'lock' in str(exc).lower():
+            raise RunInProgress('Another GLP operation holds the program lock.') from exc
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)